### PR TITLE
Fix exist operators ?| and ?&

### PR DESCRIPTION
**Pull request by MuhammadTahaNaveed**

### CODE REVIEW

**Comment by Zainab-Saad** [comment]
@MuhammadTahaNaveed 
There are a few queries that you might want to test:
```
SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?& '[true,"id"]'::agtype;
SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?| '[true,"id"]'::agtype;
SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?| '[1,"id"]'::agtype;
SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?& '[1,"id"]'::agtype;
SELECT '[{"id": 281474976710658, "label": "", "properties": {"n": 100}}]'::agtype ? 'null';
SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ? 'null';
SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge';
```

File: regress/expected/agtype.out [hunk 1]
```diff
@@ -4240,6 +4276,48 @@ SELECT '{"a":null, "b":"qq"}'::agtype ?| '["a","a", "b", "b", "b"]'::agtype;
...
+ t
+(1 row)
+
+SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?| '["id",null]'::agtype;
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?| '[true,"id"]'::agtype;
```
Comment (Zainab-Saad) [inline regress/expected/agtype.out:4297]: Should true be returned here?
`SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::jsonb ?& '{true,"id"}';` returns false.

File: regress/expected/agtype.out [hunk 2]
```diff
@@ -4240,6 +4276,48 @@ SELECT '{"a":null, "b":"qq"}'::agtype ?| '["a","a", "b", "b", "b"]'::agtype;
...
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?| '[1,"id"]'::agtype;
```
Comment (Zainab-Saad) [inline regress/expected/agtype.out:4303]: Should true be returned here?
`SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::jsonb ?& '{1,"id"}';` returns false

File: regress/expected/agtype.out [hunk 3]
```diff
@@ -4332,6 +4532,144 @@ SELECT '{"a":null, "b":"qq"}'::agtype ?& '["a","a", "b", "b", "b"]'::agtype;
...
+ t
+(1 row)
+
+SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?& '["id",null]'::agtype;
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?& '[1,"id"]'::agtype;
```
Comment (Zainab-Saad) [inline regress/expected/agtype.out:4565]: Should this return true?
`SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::jsonb ?& '{1,"id"}';` returns false

File: src/backend/utils/adt/agtype_ops.c [hunk 4]
```diff
@@ -1232,12 +1231,21 @@ Datum agtype_exists_agtype(PG_FUNCTION_ARGS)
...
-    aval = string_to_agtype_value(str);
+    if (AGT_ROOT_IS_OBJECT(agt) &&
+        aval->type == AGTV_STRING)
+    {
+        v = find_agtype_value_from_container(&agt->root,
+                                            AGT_FOBJECT,
+                                            aval);
+    }
+    else if (AGT_ROOT_IS_ARRAY(agt) &&
+                !aval->type == AGTV_NULL)
```
Comment (Zainab-Saad) [inline src/backend/utils/adt/agtype_ops.c:1242]: Will this query crash the server given only these two conditions?
`SELECT '[1,2,3]'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge'::agtype;`
And what about this one?
`SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge';`

Comment (MuhammadTahaNaveed) [reply src/backend/utils/adt/agtype_ops.c:1242]: No, it wont.

```
new=# SELECT '[1,2,3]'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge'::agtype;
 ?column? 
----------
 f
(1 row)

new=# SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge';
 ?column? 
----------
 f
(1 row)
```

File: regress/expected/agtype.out [hunk 5]
```diff
@@ -4332,6 +4532,144 @@ SELECT '{"a":null, "b":"qq"}'::agtype ?& '["a","a", "b", "b", "b"]'::agtype;
...
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ?& '[true,"id"]'::agtype;
```
[the remark this answers is not in the document]
Comment (MuhammadTahaNaveed) [reply regress/expected/agtype.out:4571]: Nice catch... Thanks for pointing out 👍🏻 . I am going to fix it now.

File: regress/expected/agtype.out (hunk 3 above)
Comment (MuhammadTahaNaveed) [reply regress/expected/agtype.out:4565]: Nice catch... Thanks for pointing out 👍🏻 . I am going to fix it now.

**Pull request merged**